README.adoc

```diff
@@ -0,0 +1,51 @@
+= docker-packaging
+
+image:https://img.shields.io/docker/pulls/jenkinsciinfra/repo-proxy?label=jenkinsciinfra%repo-proxy&logo=docker&logoColor=white[link="https://hub.docker.com/r/jenkinsciinfra/repo-proxy"]
+
+This repository contains everything needed to build a docker image for caching artefacts from repo.jenkins-ci.org repository to accelerate builds.
+
+== Running the service
+
+Start a local Docker container (running in background) based on this image with:
+
+[source,bash]
+----
+docker run --rm --name local-artifact-cacher -p 80:80 --detach jenkinssciinfra/repo-proxy # should add a tag in production
+----
+
+Cached object should return a status code `200` with HEADER `X-Cache-Status` set to `HIT` or `MISS`
+
+[source,bash]
+----
+curl -I http://localhost/releases/cf/pgmann/plugins/url-auth-sso/1.0/url-auth-sso-1.0-javadoc.jar
+----
+
+Snapshot artifacts shouldn't be cached and thus return a status code `200` without HEADER `X-Cache-Status`
```

Comment:
Probably _could_ be cached, at least timestamped versions (since we are not caching `maven-metadata.xml`), but fine—rarely used. Artifacts from the `incrementals` repository definitely _can_ be cached, and it is important to verify that: while they are used similarly to snapshots, from a technical standpoint they are releases (for our purposes: once uploaded to a particular URL, will never be overwritten with different content).